# HG changeset patch
# User Rik <rik@octave.org>
# Date 1515088327 28800
#      Thu Jan 04 09:52:07 2018 -0800
# Node ID e61e26d1ede953fcfd02cf09ecbff66ffc333e84
# Parent  3fc1c8ebe5c3e7aaba05fa05265d15535064103d
doc: Improve documentation for genpath().

* load-path.cc (Fgenpath): Improve docstring by mentioning which
file folders are excluded from generated path.

diff --git a/libinterp/corefcn/load-path.cc b/libinterp/corefcn/load-path.cc
--- a/libinterp/corefcn/load-path.cc
+++ b/libinterp/corefcn/load-path.cc
@@ -2173,18 +2173,23 @@ namespace octave
 }
 
 DEFUN (genpath, args, ,
        doc: /* -*- texinfo -*-
 @deftypefn  {} {} genpath (@var{dir})
 @deftypefnx {} {} genpath (@var{dir}, @var{skip}, @dots{})
 Return a path constructed from @var{dir} and all its subdirectories.
 
+The path does not include package directories (beginning with @samp{+}),
+old-style class directories (beginning with @samp{@@}), @file{private}
+directories, or any subdirectories of these types.
+
 If additional string parameters are given, the resulting path will exclude
 directories with those names.
+@seealso{path, addpath}
 @end deftypefn */)
 {
   int nargin = args.length ();
 
   if (nargin == 0)
     print_usage ();
 
   octave_value retval;
